### ISSUE TITLE
fix dialyzer issues

### ISSUE THREAD

**Issue by andreineculau** [for-GET/jesse#1]
https://travis-ci.org/for-GET/jesse/jobs/95224229

```
jesse_error.erl:87: Function handle_data_invalid/3 has no local return
jesse_error.erl:100: Function handle_schema_invalid/2 has no local return
jesse_error.erl:109: Function handle_error/2 has no local return
jesse_error.erl:113: Fun application will fail since ErrorHandler :: fun(({'state',_,_,'undefined' | [binary()],'infinity' | 'undefined' | non_neg_integer(),'undefined' | [any()],'undefined' | fun(({'state',_,_,'undefined' | [binary()],'infinity' | 'undefined' | non_neg_integer(),'undefined' | [any()],'undefined' | fun(({'state',_,_,'undefined' | [binary()],'infinity' | 'undefined' | non_neg_integer(),'undefined' | [any()],'undefined' | fun((_) -> [any()]),atom()}) -> [any()]),atom()}) -> [any()]),atom()}) -> [any()]) is not a function of arity 3
jesse_state.erl:84: Invalid type specification for function jesse_state:get_default_schema_ver/1. The success typing is (#state{current_path::'undefined' | [binary()],allowed_errors::'infinity' | 'undefined' | non_neg_integer(),error_list::'undefined' | [any()],error_handler::'undefined' | fun((#state{current_path::'undefined' | [any()],allowed_errors::'infinity' | 'undefined' | non_neg_integer(),error_list::'undefined' | [any()],error_handler::'undefined' | fun((_) -> any()),default_schema_ver::atom()}) -> [any()]),default_schema_ver::atom()} | jesse_state:state()) -> atom()
jesse_validator_draft3.erl:304: Function wrong_type/2 has no local return
jesse_validator_draft3.erl:405: The created fun has no local return
jesse_validator_draft3.erl:405: The call lists:foldl(fun((_,_) -> none()),State::jesse_state:state(),Extras::[{atom() | binary() | [any()],_},...]) will never return since the success typing arguments are (fun((_,_) -> any()),any(),[any()])
jesse_validator_draft3.erl:492: The created fun has no local return
jesse_validator_draft3.erl:494: The call jesse_validator_draft3:check_value(Index::number(),Item::any(),Items::any(),CurrentState::any()) will never return since it differs in the 1st argument from the success typing arguments: (binary(),any(),any(),jesse_state:state())
jesse_validator_draft3.erl:541: The created fun has no local return
jesse_validator_draft3.erl:546: The call jesse_validator_draft3:check_value(Index::number(),Item::any(),Schema::any(),NewState::jesse_state:state()) will never return since it differs in the 1st argument from the success typing arguments: (binary(),any(),any(),jesse_state:state())
jesse_validator_draft3.erl:848: The call jesse_validator_draft3:check_value('extends',Value::any(),Extends::any(),jesse_state:state()) will never return since it differs in the 1st argument from the success typing arguments: (binary(),any(),any(),jesse_state:state())
jesse_validator_draft3.erl:939: Function handle_data_invalid/3 has no local return
jesse_validator_draft3.erl:943: Function handle_schema_invalid/2 has no local return
jesse_validator_draft3.erl:963: Function remove_last_from_path/1 will never be called
```

**Comment by andreineculau** [comment]
closing in favour of doing this after #5
